crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -529,10 +534,41 @@ impl RpcServerConfig {
     pub fn ipc(config: IpcServerBuilder) -> Self {
         Self::default().with_ipc(config)
     }
-
     /// Configures the http server
-    pub fn with_http(mut self, config: ServerBuilder) -> Self {
-        self.http_server_config = Some(config.http_only());
+    pub fn with_http(mut self, config: ServerBuilder, domains: Option<&String>) -> Self {
+        if let Some(domains) = domains {
+            match domains.as_str() {
+                "*" => {
+                    let cors = CorsLayer::new()
+                        .allow_methods([Method::GET, Method::POST])
+                        .allow_origin(Any)
+                        .allow_headers(Any);
+
+                    let middleware = tower::ServiceBuilder::new().layer(cors);
+                    self.http_cors_domain_server_config =
+                        Some(config.set_middleware(middleware).http_only());
+                }
```

Comment:
all of this needs to be set when the server is started

Reply:
yeah @Tirthnp what you want here is to just set the config variable, and then in the build/start function you actually consume them